libazureinit/src/provision/user.rs

```diff
@@ -123,10 +127,39 @@ fn useradd(user: &User) -> Result<(), Error> {
     Ok(())
 }
 
+fn add_user_for_passwordless_sudo(
+    username: String,
```

Comment:
Since we're going to use this as part of a format string, we don't need to own the value. Instead, we can take a read-only reference to the string and avoid the need to clone it in the caller. You can change `String` to `&str` here and then you should be able to replace the `user.name.clone()` call with `user.name.as_str()`

Reply:
Done